### ISSUE TITLE
Configure TransactionPool via ProtocolSchedule

### ISSUE THREAD

**Issue by garyschulte** [hyperledger/besu#2551]
We want to configure the transaction pool and applicable handlers (e.g. TransactionPoolReplacementHandler) via ProtocolSchedule.

**Comment by garyschulte** [comment]
The challenge is that TransactionPool and related classes are in ethereum.eth, and ProtocolSchedule is in ethereum.core.  Ethereum.eth has a dependency on core, and we need to avoid circular dependencies amongst submodules.

One way to solve this is to move TransactionPool related code into the core gradle subproject.  

Another option is to move ProtocolSchedule into a separate config submodule so we can have dependencies on other submodules.

IMO having ProtocolSchedule in core is going to require that anything we want to instance or directly configure to be in core, since nearly every submodule has an upstream dependency on core.

**Comment by jflo** [comment]
The difference between ethereum.core and ethereum.eth is unclear to me. Does anything other than ethereum.eth depend on core? I assume it does, and those things do not want to also depend on ethereum.eth. I think the real question to be answered is "are these submodules useful on their own"? Or phrased differently "is it useful to track different versions of a given submodule"?

Wrinkle: The transaction pool itself could house transactions that are in different protocol schedules. It is possible that the pool has no protocol version specific behaviors, but the operation of replacing transactions might in fact be version specific.

One of the common conditionals I saw a lot of while in there was differences in behavior for local vs. remote transactions. I guess we need to add transport mechanism to the list of factors that adjust our behavior.

**Comment by non-fungible-nelson** [comment]
@fab-10 - any input here with new generation tx pool? Ok to close or a small refactor that would improve behavior?

**Comment by fab-10** [comment]
Was not aware of this one, and the new tx pool does not change the behavior, but I need to know which is the need to have this.

@garyschulte is the need for this described in this comment?
```
  // There's a small edge case here we could encounter.
  // When we pass an upgrade block that has a new transaction type, we start allowing transactions
  // of that new type into our pool.
  // If we then reorg to a block lower than the upgrade block height _and_ we create a block, that
  // block could end up with transactions of the new type.
  // This seems like it would be very rare but worth it to document that we don't handle that case
  // right now.
```

**Comment by garyschulte** [comment]
> Was not aware of this one, and the new tx pool does not change the behavior, but I need to know which is the need to have this.
> 
> @garyschulte is the need for this described in this comment?
> ```
>   // There's a small edge case here we could encounter.
>   // When we pass an upgrade block ... 
> ```

Yes.  IIRC the intent was to modularize/configure the transaction pool rather than having a monolithic implementation.  

A txpool that changes with protocol schedule is a bigger ask however.  We could more easily have configurable transaction pools which are monolithic across protocol schedulesl changes, but configured per chain rather than per protocolspec.

IMO this issue is of dubious value presently
